lib/liquid/variable.rb

```diff
@@ -31,6 +31,8 @@ def initialize(markup, parse_context)
       @line_number   = parse_context.line_number
 
       parse_with_selected_parser(markup)
+
+      Optimizer.optimize_variable(self)
```

Comment:
Note that this will be skipped when compiling with liquid-c if it passes strict parsing without a syntax error.

Reply:
can you elaborate?

Reply:
liquid-c won't allocate the Variable instance at all.

I guess we'll probably want to duplicate the optimizer inside C. It's probably not practical to have a single implementation for both.